xbmc/input/InputManager.cpp

```diff
@@ -389,10 +430,372 @@ bool CInputManager::ProcessJoystickEvent(int windowId, const std::string& joysti
+  if (!action.GetName().empty())
+  {
+    std::string builtInFunction;
+    std::vector<std::string> params;
+    CUtil::SplitExecFunction(action.GetName(), builtInFunction, params);
+    StringUtils::ToLower(builtInFunction);
+
+    // should this button be handled normally or just cancel the screensaver?
+    if (builtInFunction == "powerdown"
+        || builtInFunction == "reboot"
+        || builtInFunction == "restart"
+        || builtInFunction == "restartapp"
+        || builtInFunction == "suspend"
+        || builtInFunction == "hibernate"
+        || builtInFunction == "quit"
+        || builtInFunction == "shutdown")
+    {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+bool CInputManager::ExecuteInputAction(const CAction &action)
+{
+  bool bResult = false;
+
+  // play sound before the action unless the button is held,
+  // where we execute after the action as held actions aren't fired every time.
+  if (action.GetHoldTime())
+  {
+    bResult = g_application.OnAction(action);
+    if (bResult)
+      g_audioManager.PlayActionSound(action);
+  }
+  else
+  {
+    g_audioManager.PlayActionSound(action);
+    bResult = g_application.OnAction(action);
+  }
+  return bResult;
+}
+
+int CInputManager::ExecuteBuiltin(const std::string& execute, const std::vector<std::string>& params)
+{
+#if defined(HAS_LIRC) || defined(HAS_IRSERVERSUITE)
+  if (execute == "lirc.stop")
+  {
+    m_RemoteControl.Disconnect();
+    m_RemoteControl.setUsed(false);
+  }
+  else if (execute == "lirc.start")
+  {
+    m_RemoteControl.setUsed(true);
+    m_RemoteControl.Initialize();
+  }
+  else if (execute == "lirc.send")
+  {
+    std::string command;
+    for (int i = 0; i < (int)params.size(); i++)
+    {
+      command += params[i];
+      if (i < (int)params.size() - 1)
+        command += ' ';
+    }
+    m_RemoteControl.AddSendCommand(command);
+  }
+  else
+    return -1;
+#endif
+  return 0;
+}
+
+void CInputManager::SetMouseActive(bool active /* = true */)
+{
+  m_Mouse.SetActive(active);
+}
+
+void CInputManager::SetMouseEnabled(bool mouseEnabled /* = true */)
+{
+  m_Mouse.SetEnabled(mouseEnabled);
+}
+
+bool CInputManager::IsMouseActive()
+{
+  return m_Mouse.IsActive();
+}
+
+MOUSE_STATE CInputManager::GetMouseState()
+{
+  return m_Mouse.GetState();
+}
+
+MousePosition CInputManager::GetMousePosition()
+{
+  return m_Mouse.GetPosition();
+}
+
+void CInputManager::SetMouseResolution(int maxX, int maxY, float speedX, float speedY)
+{
+  m_Mouse.SetResolution(maxX, maxY, speedX, speedY);
+}
+
+void CInputManager::SetMouseState(MOUSE_STATE mouseState)
+{
+  m_Mouse.SetState(mouseState);
+}
+
+bool CInputManager::IsRemoteControlEnabled()
+{
+#if defined(HAS_LIRC) || defined(HAS_IRSERVERSUITE)
+  return m_RemoteControl.IsInUse();
+#else
+  return false;
+#endif
+}
+
+void CInputManager::EnableRemoteControl()
+{
+#if defined(HAS_LIRC) || defined(HAS_IRSERVERSUITE)
+  m_RemoteControl.setUsed(true);
```

Comment:
This looks like a change in behavior and is causing issues on Linux. It's now trying to init on every `ProcessSlow()` call.